src/dash-playlist-loader.js

```diff
@@ -0,0 +1,173 @@
+      throw new Error('Cannot switch media playlist from ' + this.state);
+    }
+
+    // find the playlist object if the target playlist has been specified by URI
+    if (typeof playlist === 'string') {
+      if (!this.master.playlists[playlist]) {
+        throw new Error('Unknown playlist URI: ' + playlist);
+      }
+      playlist = this.master.playlists[playlist];
+    }
+
+    const mediaChange = !this.media_ || playlist.uri !== this.media_.uri;
+
+    this.state = 'HAVE_METADATA';
+    this.media_ = playlist;
+
+    // trigger media change if the active media has been updated
+    if (mediaChange) {
+      this.trigger('mediachanging');
+      this.trigger('mediachange');
+    }
+    return;
+  }
+
+  pause() {
+    this.stopRequest();
+    if (this.state === 'HAVE_NOTHING') {
+      // If we pause the loader before any data has been retrieved, its as if we never
+      // started, so reset to an unstarted state.
+      this.started = false;
+    }
+  }
+
+  load() {
+    // because the playlists are internal to the manifest, load should either load the
+    // main manifest, or do nothing but trigger an event
+    if (!this.started) {
+      this.start();
+      return;
+    }
+
+    this.trigger('loadedplaylist');
+  }
+
+  start() {
+    this.started = true;
+
+    // request the specified URL
+    this.request = this.hls_.xhr({
+      uri: this.srcUrl,
+      withCredentials: this.withCredentials
+    }, (error, req) => {
+      // disposed
+      if (!this.request) {
+        return;
+      }
+
+      // clear the loader's request reference
+      this.request = null;
+
+      if (error) {
+        this.error = {
+          status: req.status,
+          message: 'DASH playlist request error at URL: ' + this.srcUrl,
+          responseText: req.responseText,
+          // MEDIA_ERR_NETWORK
+          code: 2
+        };
+        if (this.state === 'HAVE_NOTHING') {
+          this.started = false;
+        }
+        return this.trigger('error');
+      }
+
+      this.master = mpdParser.parse(req.responseText, this.srcUrl);
+      this.master.uri = this.srcUrl;
+
+      this.state = 'HAVE_MASTER';
+
+      // TODO mediaSequence will be added in mpd-parser
+      this.master.playlists.forEach((playlist) => {
+        playlist.mediaSequence = 0;
+      });
+      for (let groupKey in this.master.mediaGroups.AUDIO) {
+        for (let labelKey in this.master.mediaGroups.AUDIO[groupKey]) {
+          this.master.mediaGroups.AUDIO[groupKey][labelKey].playlists.forEach(
+            (playlist) => {
+              playlist.mediaSequence = 0;
+            });
+        }
+      }
+
+      // set up phony URIs for the playlists since we won't have external URIs for DASH
+      // but reference playlists by their URI throughout the project
+      for (let i = 0; i < this.master.playlists.length; i++) {
+        const phonyUri = `placeholder-uri-${i}`;
+
+        this.master.playlists[i].uri = phonyUri;
```

Comment:
should also add `resolvedUri` here, quite a few places outside of this class use `resolvedUri` instead of `uri`

Reply:
maybe not quite a few since media-groups has source type checking now, but there is a spot in MPC and external plugins may look for it

Reply:
I forgot, it actually will get added in `setupMediaPlaylists`